pkg/applicationcontroller/applicationcontroller_suite_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Authors of spidernet-io
+// SPDX-License-Identifier: Apache-2.0
+
+package applicationcontroller
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	kruiseapi "github.com/openkruise/kruise-api"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	spiderpoolv2beta1 "github.com/spidernet-io/spiderpool/pkg/k8s/apis/spiderpool.spidernet.io/v2beta1"
+	"github.com/spidernet-io/spiderpool/pkg/reservedipmanager"
+	"github.com/spidernet-io/spiderpool/pkg/subnetmanager"
+)
+
+func TestApplicationcontroller(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Applicationcontroller Suite", Label("ApplicationController", "unitest"))
+}
+
+var scheme *runtime.Scheme
+var subnetAppControllerConfig SubnetAppControllerConfig
+
+type subnetApplicationController struct {
+	*SubnetAppController
+	deploymentStore  cache.Store
+	replicasSetStore cache.Store
+	daemonSetStore   cache.Store
+	statefulSetStore cache.Store
+	jobStore         cache.Store
+	cronJobStore     cache.Store
+}
+
+var _ = BeforeSuite(func() {
+	scheme = runtime.NewScheme()
+	err := spiderpoolv2beta1.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+	err = kruiseapi.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	subnetAppControllerConfig = SubnetAppControllerConfig{
+		EnableIPv4:                    true,
+		EnableIPv6:                    true,
+		AppControllerWorkers:          3,
+		MaxWorkqueueLength:            5000,
+		WorkQueueMaxRetries:           10,
+		WorkQueueRequeueDelayDuration: -1 * time.Second,
+		LeaderRetryElectGap:           0,
+	}
+})
+
+func newController() (*subnetApplicationController, error) {
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+	apiReader := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	reservedIPManager, err := reservedipmanager.NewReservedIPManager(fakeClient, apiReader)
+	if nil != err {
+		return nil, err
+	}
+	subnetManager, err := subnetmanager.NewSubnetManager(fakeClient, apiReader, reservedIPManager)
+	if nil != err {
+		return nil, err
+	}
+
+	appController, err := NewSubnetAppController(fakeClient, apiReader, subnetManager, subnetAppControllerConfig)
+	if nil != err {
+		return nil, err
+	}
+	return &subnetApplicationController{
+		SubnetAppController: appController,
+	}, nil
+}
```

Comment:
> the code review
> 
> First, the code looks well organized and is easy to read. The use of comments make it easier to understand. The code follows the Go programming language standards, with proper indentation and clear variable names.
> 
> However, there are a few potential risks in the code that may need to be addressed. First, there is no error handling when creating the SubnetAppController. It could be beneficial to add some error handling when creating the SubnetAppController, in order to better handle any unexpected errors.
> 
> Second, the MaxWorkQueueLength is set to 5000. It could be beneficial to set this value to a lower number in order to better handle any unexpected load.
> 
> Finally, the WorkQueueRequeueDelayDuration is set to -1 * time.Second. This could potentially lead to an infinite loop if the controller encounters an error while processing a request. It would be beneficial to set this value to a more appropriate value in order to avoid this issue.
> 
> Overall, the code looks good and there are no major issues. With the addition of some error handling and tweaking of certain values, the code can be improved to ensure that it is more robust.

hello, guys.